web-client/src/app/views/delete-user/delete-user.page.ts

```diff
@@ -0,0 +1,260 @@
+  templateUrl: './delete-user.page.html',
+  styleUrls: ['./delete-user.page.scss'],
+})
+export class DeleteUserPage implements OnInit {
+  @Input() isPinEntryOpen = false;
+
+  /** True if balances are in the process of being updated */
+  @Input() balancesIsLoading = false;
+
+  actionItems: Array<SendFundsActionItem> = [
+    {
+      title: 'Scan a QR code',
+      icon: faQrcode,
+      action: 'presentScanner',
+    },
+    {
+      title: 'Enter address manually',
+      icon: faKeyboard,
+      action: 'presentAddressModal',
+    },
+    // {
+    //   title: 'Share my wallet address',
+    //   icon: faLink,
+    //   disabled: true,
+    // },
+  ];
+
+  addressForm: FormGroup;
+
+  /** Active wallet's balances. */
+  balances: Observable<AssetAmount[]> = this.sessionQuery.allBalances;
+
+  hasCamera?: boolean;
+
+  /** @see validatedAddress */
+  address?: string;
+
+  constructor(
+    private navCtrl: NavController,
+    private modalCtrl: ModalController,
+    private loadingController: LoadingController,
+    public sessionService: SessionService,
+    public sessionQuery: SessionQuery,
+    public notification: SwalHelper,
+    public sessionAlgorandService: SessionAlgorandService,
+    public sessionXrplService: SessionXrplService
+  ) {
+    this.addressForm = new FormGroup({
+      address: new FormControl('', [Validators.required, addressValidator]),
+    });
+  }
+
+  get validatedAddress(): string | undefined {
+    const trimmed = this.address?.trim();
+    return trimmed === '' ? undefined : trimmed;
+  }
+
+  async ngOnInit(): Promise<void> {
+    await this.refreshWalletData();
+  }
+
+  async openScanner(): Promise<void> {
+    await handleScan(
+      this.modalCtrl,
+      this.notification.swal,
+      this.confirmAddress
+    );
+  }
+
+  /** User clicked to confirm address: show PIN entry. */
+  async confirmAddress(): Promise<void> {
+    if (this.validatedAddress !== undefined) {
+      this.showPinEntryModal();
+    } else {
+      await this.notification.swal.fire({
+        icon: 'warning',
+        title: 'Invalid Address',
+        text: 'Please input a valid wallet address',
+      });
+    }
+  }
+
+  /** Show the PIN entry modal. */
+  showPinEntryModal(): void {
+    this.isPinEntryOpen = true;
+  }
+
+  dismiss(success: boolean, address?: string) {
+    this.modalCtrl.dismiss({
+      success,
+      address,
+    });
+  }
+
+  onSubmit() {
+    this.addressForm.markAllAsTouched();
+    if (this.addressForm.valid) {
+      const formControl = checkClass(
+        this.addressForm.controls.address,
+        FormControl
+      );
+      const address = trimmedValue(formControl);
+      this.dismiss(true, address);
+    }
+  }
+
+  async onRefresh(): Promise<void> {
+    await withLoadingOverlayOpts(
+      this.loadingController,
+      { message: 'Refreshing…' },
+      async () => await this.refreshWalletData()
+    );
+  }
+
+  async refreshWalletData(): Promise<void> {
```

Comment:
In the future the code duplicated between the delete-user page and the wallet page should be moved to a service or other components that can be shared. I think this should be added as a backlog task. @Claire-Bear

Reply:
@longtomjr and @Claire-Bear, I've added [this task](https://app.clickup.com/t/2f9a4mq) on the board and can be tackled after milestone 2